### PR TITLE
Fixes in cleaner

### PR DESCRIPTION
**Pull request by robertbaldyga**

### CODE REVIEW

File: src/cleaning/acp.c [hunk 1]
```diff
@@ -652,7 +652,6 @@ static void _acp_flush(struct acp_context *acp)
 		.cmpl_fn = _acp_flush_end,
 		.lock_cacheline = false,
 		.lock_metadata = true,
-		.do_sort = false,
 		.cmpl_queue = true,
 		.io_queue = cache->cleaner.io_queue,
 	};
```
Comment (rafalste) [inline src/cleaning/acp.c:656]: ocf_cleaner_sort_flush_data() before the ocf_cleaner_do_flush_data_async() here as well?

Comment (rafalste) [reply src/cleaning/acp.c:656]: And how about the one in ocf/src/mngt/ocf_mngt_flush.c:366 ?

File: src/utils/utils_cleaner.c [hunk 2]
```diff
@@ -722,14 +722,41 @@ static int _ocf_cleaner_fire_cache(struct ocf_request *req)
...
+
 static int _ocf_cleaner_do_fire(struct ocf_request *req, uint32_t count)
 {
 	int result;
 
 	/* Set counts of cache IOs */
 	env_atomic_set(&req->req_remaining, count);
-
-	req->engine_handler = _ocf_cleaner_fire_cache;
+
```
Comment (rafalste) [inline src/utils/utils_cleaner.c:752]: lost tab

File: src/ocf_lru.c [hunk 3]
```diff
@@ -643,21 +629,25 @@ void ocf_lru_clean(ocf_cache_t cache, struct ocf_user_part *user_part,
...
+	for (i = 0; i < count; i++) {
+		entries[i].cache_line = lru_iter_cleaning_next(&iter);
+		if (entries[i].cache_line == end_marker)
 			break;
-		i++;
+		ocf_metadata_get_core_info(cache, entries[i].cache_line,
+				&entries[i].core_id, &entries[i].core_line);
 	}
-	while (i < OCF_EVICTION_CLEAN_SIZE)
-		cline[i++] = end_marker;
```
[the remark this answers is not in the document]
Comment (robertbaldyga) [reply src/ocf_lru.c:654]: Yes, we pass the exact count now. We need `end_marker` only on the first invalid cache line for unlocking (which is done in line 632).

File: src/cleaning/acp.c (hunk 1 above)
Comment (robertbaldyga) [reply src/cleaning/acp.c:656]: In ACP there is no need for sorting, because chunks operate on core lines, so they are sorted by definition.

Comment (robertbaldyga) [reply src/cleaning/acp.c:656]: > And how about the one in ocf/src/mngt/ocf_mngt_flush.c:366 ?

In management flush all the flush containers are sorted by one huge global sort in src/mngt/ocf_mngt_flush.c:491.

File: src/utils/utils_cleaner.c (hunk 2 above)
Comment (robertbaldyga) [reply src/utils/utils_cleaner.c]: Fixed.

**Review by mmichal10 — APPROVED** [review]

**Pull request merged**